### DISCUSSION TITLE
[Build call without name][Hw3]

### DISCUSSION THREAD

**Discussion by kopter0** [prosyslab-classroom/cs348-information-security#742] [Q&A]
`__dbz_sanitizer__` has void as return type
When I inserted it in with `build_call` , I get error message 

> instructions returning void cannot have a name

I have checked API documentation several times, and couldnt find `build_call` without name option
How to insert instruction without setting name for it?

**Comment by pacokwon** [comment]
Just putting an empty string did it for me, although I'm not sure if this is intended behavior.

**Comment by pacokwon** [comment]
I think this is the relevant code that gets executed on empty string and the behavior seems to be intended. Maybe it could be helpful!
https://github.com/llvm/llvm-project/blob/a76df78470d7994f73df0353225cbddc463cce63/llvm/lib/IR/Value.cpp#L308-L315

**Comment by Standchen** [comment]
I think the 3rd argument of `build_call` denotes the name of resulting variable.
If you look into LLVM assembly in .ll files, there's something you can see like:
```
%**div** = sdiv i32 30, %a, !dbg !18
%**div1** = sdiv i32 40, %a, !dbg !21
%**inc19** = add nsw i32 5, 1, !dbg !46
```

From these, I think LHS (`div`, `div1`, `inc19`) is the name of somewhat variables in LLVM assembly (maybe I was wrong, it could be the name of variables in LLVM IR, or whatever), and the 3rd argument of `build_call` gives us room to choose such name.

But since instructions(like function calls to `__dbz_sanitizer__`, `__coverage__`) returning void literally return nothing, we can't just name it. Just leaving the 3rd argument empty string ("") worked fine in my case.

**Comment by KihongHeo** [comment]
@SpieleMann is right. It is okay to use `""`.

**Comment by kopter0** [comment]
Thx
